src/main/java/org/jboss/modules/JDKSpecific.java

```diff
@@ -64,7 +63,9 @@
     // === the actual JDK-specific API ===
 
     static Class<?> getCallingUserClass() {
-        return STACK_WALKER.walk(stream -> stream.skip(1).findFirst().get().getDeclaringClass());
+        return STACK_WALKER.walk(stream -> stream.skip(1)
+                .filter(s -> !s.getDeclaringClass().equals(org.jboss.modules.Module.class))
+                .findFirst().get().getDeclaringClass());
```

Comment:
What case does this solve? I didn't run across any failures that might relate to this.